frontend/src/pages/signUp/SignUp.tsx

```diff
@@ -20,20 +21,36 @@ const SignUp = () => {
 
   const { isDirty, isValid } = formState;
 
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [editingPass, setEditingPass] = useState<boolean>(false);
+
   const navigate = useNavigate();
 
   const onSubmit: SubmitHandler<FormData> = async (data) => {
     try {
       await createUserWithEmailAndPassword(auth, data.email, data.password);
+      setErrorMessage('');
       navigate('/login');
-    } catch (err) {
-      console.error(err);
+    } catch (e: any) {
+      if (e.message === 'Firebase: Error (auth/email-already-in-use).') {
+        setErrorMessage(`Email address '${data.email}' is already in use `);
```

Comment:
not a good idea to do comparison on the message like that, if firebase decides to change the returned error your code breaks. There must be a better way to detect the error type.